Make bytes indexing and iteration return/yield uint(8)

e-kayrakli opened the pull request:
This PR addresses the third design question in #13901 by making:

- `bytes.this` return `uint(8)` instead of `bytes`
- `bytes.these` yield `uint(8)` instead of `bytes`

Also adds `item` and `items` to `bytes` and `string` that returns/yields values
of the same type for easier generic programming.

The main motivation for this change is symmetry with Python, and `bytes` being a
more array-like type than `string`. A more detailed discussion for this decision
can be found in #13901.

Test:
- [x] standard local
- [x] standard gasnet

mppf commented on modules/internal/String.chpl:
```diff
@@ -977,6 +977,32 @@ module String {
+
+    /*
+      Iterates over the string character by character. (A synonym for
+      :iter:`chars`)
```
this should say that it yields strings with 1 codepoint each

e-kayrakli commented:
The last update changes the proc/iter names from `char` to `item` as agreed upon in #13901. Also addresses the feedback on doc. I will run some more tests to be sure and merge:

- [ ] local
- [ ] gasnet
